btr-web/btr-main-app/utils/validation.ts

```diff
@@ -72,12 +72,12 @@ export function validateBirthDate (birthDate: any): boolean {
  * @param taxData the form data
  */
 export function validateTaxNumberInfo (
-  taxData: { taxNumber: string | null, hasTaxNumber: boolean | null },
+  taxData: { taxNumber?: string, hasTaxNumber?: boolean },
   ctx: RefinementCtx
 ): never {
   const t = useNuxtApp().$i18n.t
-  if (taxData.hasTaxNumber) {
-    if (!checkSpecialCharacters(taxData.taxNumber)) {
+  if (taxData.hasTaxNumber && taxData.taxNumber) {
+    if (!checkSpecialCharacters(taxData.taxNumber!)) {
```

Comment:
for this and below: is the extra '!' still necessary? You updated the typing of the function  to include 'undefined' as well so I don't think you need it right?